Core.ApplicationServices/SystemUsage/Write/ItSystemUsageWriteService.cs

```diff
@@ -135,6 +135,35 @@ public class ItSystemUsageWriteService : IItSystemUsageWriteService
             return creationResult;
         }
 
+        public Result<ExternalReference, OperationError> AddExternalReference(Guid usageUuid, ExternalReferenceProperties externalReferenceProperties)
+        {
+            return GetUsageAndAuthorizeWriteAccess(usageUuid)
+                .Bind(usage => _referenceService.AddReference(usage.Id, ReferenceRootType.SystemUsage, externalReferenceProperties));
+        }
+
+        public Result<ExternalReference, OperationError> UpdateExternalReference(Guid usageUuid, Guid externalReferenceUuid, ExternalReferenceProperties externalReferenceProperties)
+        {
+            return GetUsageAndAuthorizeWriteAccess(usageUuid)
+                .Bind(usage => _referenceService.UpdateReference(usage.Id, ReferenceRootType.SystemUsage, externalReferenceUuid, externalReferenceProperties));
```

Comment:
I'm not 100% sure if creating was the right approach, or if I should have used the batch update method we already have

Reply:
this is fine - you already have the methods in the service, so we use that.